### PR TITLE
New restore role

### PR DESCRIPTION
**Pull request by t-woerner**
There is a new restore role in the roles folder:

    roles/iparestore

This role allows to restore an IPA server locally and from the controller
and also to copy a backup from the controller to the server.

Here is the documentation for the role:

    roles/ipabackup/README.md

New example playbooks have been added:

    playbooks/restore-server.yml
    playbooks/restore-server-from-controller.yml
    playbooks/copy-backup-to-server.yml

The vars are copied form the ipaserver role to be able to install the
needed packages. The vars should be moved into a common role to be consumed
by ipaserver, ipareplica and iparestore roles later on.

### CODE REVIEW

File: roles/iparestore/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+iparestore role
+==============
+
+Description
+-----------
+
+This roles allows to restore an IPA server locally and from the controller and also to copy a backup from the controller to the server.
```
Comment (rjeffman) [inline roles/iparestore/README.md:7]: I believe it is "This role" (single).

**Review by rjeffman — CHANGES_REQUESTED** [review]
Except for a minor documentation issue, looks good.

File: roles/iparestore/README.md (hunk 1 above)
Comment (t-woerner) [reply roles/iparestore/README.md]: Fixed

**Comment by t-woerner** [comment]
I have added support for the NTP service in the firewall for IPA 4.6.

**Comment by amore17** [comment]
Getting error `fatal: [master.ipa.test]: FAILED! => {"msg": "Unexpected templating type error occurred on ({{ iparestore_name | regex_search('.*_(ipa-.+)','\\\\1') | first }}): 'NoneType' object is not iterable"}`
With `iparestore_from_controller: yes`
`state: present`

File: roles/iparestore/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+iparestore_no_logs | Do not restore log files from the backup, bool (default: `no`) | no
+iparestore_log_file | Log to the given file on server, string | no
+state | `present` to make a new restore, `copied` to copy a restore from the server to the controller. string (default: `present`) | yes
+
+Special Variables
+-----------------
+
+Variable | Description | Required
+-------- | ----------- | --------
+iparestore_from_controller | Copy backup to controller, restore if `state: present`, copy backup to server if `state: copied`, bool (default: `no`) | no
```
Comment (amore17) [inline roles/iparestore/README.md:144]: `Copy backup to controller` is quite confusing in description.
copying backup to controller is prerequisite right ?

Comment (t-woerner) [reply roles/iparestore/README.md:144]: It should have been `copy backup to node` maybe also `copy backup from controller to node`

**Comment by t-woerner** [comment]
There is a new PR with a combined backup and restore role: #430

**Pull request closed without merging**